src/main/java/com/conveyal/r5/streets/EdgeStore.java

```diff
@@ -728,6 +729,32 @@ public boolean canTurnFrom(StreetRouter.State s0, StreetRouter.State s1,
             return true;
         }
 
+        /**
+         * Cut the packed list of intermediate coordinates in two after specified segment
+         * @return List with first element int[] of coordinates before end of specified segment, and second element
+         * int[] of coordinates after end of specified segment.
+         */
+        public ArrayList<int[]> splitGeometryAfter(int segment) {
+            // Original packed coordinates of edge
+            int[] original = geometries.get(pairIndex);
+            ArrayList<int[]> geoms = new ArrayList<>();
```

Comment:
Returning a List works from a practical point of view, but it feels a bit like it can have an arbitrary number of elements and can be modified / extended. We have pair and tuple types defined (`P2` and `T2`), so maybe this could return a `P2<int[]>`.